forms-flow-web/src/components/Modeller/ModelEditorHook.js

```diff
@@ -0,0 +1,313 @@
+
+    const { t } = useTranslation();
+
+    const dispatch = useDispatch();
+    const diagramXML = useSelector((state) => state.process.processDiagramXML);
+    const [bpmnModeller, setBpmnModeller] = useState(null);
+
+    const containerRef = useCallback((node) => {
+      if (node !== null) {
+        setBpmnModeller(new BpmnModeler ({ 
+          container: "#container",
+          propertiesPanel: {
+            parent: '#js-properties-panel'
+          },
+          additionalModules: [
+            BpmnPropertiesPanelModule,
+            BpmnPropertiesProviderModule,
+            CamundaPlatformPropertiesProviderModule,
+            CamundaExtensionModule
+          ],
+          moddleExtensions: {
+            camunda: camundaModdleDescriptors
+          }
+        }));
+      }
+    }, []);
+
+    useEffect(() => {
+      if (bpmnModeller) {
+        bpmnModeller.on("import.done", (event) => {
+          const { error } = event;
+          if (error) {
+            console.log("bpmnViewer error >", error);
+          }
+        });
+      }
+      return () => {
+        bpmnModeller && bpmnModeller.destroy();
+      };
+    }, [bpmnModeller]);
+
+    useEffect(() => {
+      if (processKey) {
+        dispatch(setProcessDiagramLoading(true));
+        dispatch(fetchDiagram(processKey, tenant));
+      } else {
+        dispatch(setProcessDiagramLoading(false));
+      }
+      return () => {
+        dispatch(setProcessDiagramLoading(true));
+        dispatch(setProcessDiagramXML(""));
+      };
+    }, [processKey, tenant, dispatch]);
+
+    useEffect(() => {
+      if (processInstanceId) {
+        dispatch(getProcessActivities(processInstanceId));
+      }
+      return () => {
+        dispatch(setProcessActivityData(null));
+      };
+    }, [processInstanceId, dispatch]);
+
+    useEffect(() => {
+      if (diagramXML && bpmnModeller) {
+        bpmnModeller.importXML(diagramXML)
+        .then(({ warnings }) => {
+          if (warnings.length) {
+            console.log("Warnings", warnings);
+          }
+        })
+        .catch((err) => {
+          console.log("error", err);
+        });
+      }
+
+    }, [diagramXML, bpmnModeller]);
+
+    async function exportDiagram() {
+      try {
+        // Convert diagram to xml
+        const { xml } = await bpmnModeller.saveXML();
+        // Deploy to Camunda
+        deployBPMN(xml);
+      } catch (err) {
+        console.error(err);
+      }
+    }
+
+    const createBpmnForm = (xml) => {
+      const form = new FormData();
+
+      const names = getDeploymentNames();
+
+      // Deployment Name
+      form.append('deployment-name', names.deploymentName);
+      // Deployment Source
+      form.append('deployment-source', 'Camunda Modeler');
+      // Tenant ID
+      if (tenant) {
+        form.append('tenant-id', tenant);
+      }
+      // Make sure that we do not re-deploy already existing deployment
+      form.append('enable-duplicate-filtering', 'true');
```

Comment:
looks good , Post api itself is used with param for redeploy existing workflow right ?

Reply:
I'm not sure if the post api uses a param to check for an existing workflow. I believe we still want this ('enable-duplicate-filtering', 'true') to be here as it compares the xml between the passed model and the models in the database. If the model has the same name but has different xml, then it still gets deployed and this latest deployed model will get rendered and used in forms flow